File: exercises/python_playwright/Dynamic_ID/dynamic_id_playwright.py
from playwright.sync_api import Page


class DynamicID:
    def __init__(self, page: Page):
        """Store the Playwright page and the dynamic button locator."""
        self.page = page
        self.button = page.get_by_role("button", name="Button with Dynamic ID")


    def open(self) -> None:
        """Open the Dynamic ID exercise page."""
        self.page.goto("http://uitestingplayground.com/dynamicid")


    def click_dynamic_button(self) -> None:
        """Click the button with the dynamic ID."""
        self.button.click()

File: exercises/javascript_playwright/dynamic-id/dynamicIdPlaywright.js
export class DynamicIdPage {
    /**
     *
     * @param {import('@playwright/test').Page} page
     */
    constructor(page) {
        this.page = page;
        this.button = page.getByRole("button", {
            name: "Button with Dynamic ID",
        });
    }

    async openExercisePage() {
        await this.page.goto("http://uitestingplayground.com/dynamicid");
    }

    async clickButton() {
        await this.button.click();
    }
}
